pkg/util/credentials.go

```diff
@@ -18,22 +18,54 @@ package util
 
 import (
 	"fmt"
+	"io/ioutil"
+	"os"
 )
 
 const (
-	credUserID   = "userID"
-	credUserKey  = "userKey"
-	credAdminID  = "adminID"
-	credAdminKey = "adminKey"
-	credMonitors = "monitors"
+	credUserID           = "userID"
+	credUserKey          = "userKey"
+	credAdminID          = "adminID"
+	credAdminKey         = "adminKey"
+	credMonitors         = "monitors"
+	tmpKeyFileLocation   = "/tmp/csi/keys"
+	tmpKeyFileNamePrefix = "keyfile-"
 )
 
 type Credentials struct {
-	ID  string
-	Key string
+	ID      string
+	KeyFile string
 }
 
-func getCredentials(idField, keyField string, secrets map[string]string) (*Credentials, error) {
+func storeKey(key string) (string, error) {
```

Comment:
is there any performance impact for creating and delete a file on each request? 

for the same userID and password are we creating a new file each time? if yes can we avoid it?

Reply:
> is there any performance impact for creating and delete a file on each request?

I have not measured the performance impact due to this, do you want me to?

> 
> for the same userID and password are we creating a new file each time? if yes can we avoid it?

I would say, we should avoid credential caching in any form, unless we have no other way around it. Thus, my answer is we cannot avoid it.

Reply:
@humblec any comments?